pilot/pkg/model/context.go

```diff
@@ -852,34 +860,31 @@ func (node *Proxy) SetWorkloadLabels(env *Environment) {
 	}
 }
 
-// DiscoverIPVersions discovers the IP Versions supported by Proxy based on its IP addresses.
-func (node *Proxy) DiscoverIPVersions() {
-	for i := 0; i < len(node.IPAddresses); i++ {
-		addr := net.ParseIP(node.IPAddresses[i])
-		if addr == nil {
-			// Should not happen, invalid IP in proxy's IPAddresses slice should have been caught earlier,
-			// skip it to prevent a panic.
-			continue
-		}
-		if node.GlobalUnicastIP == "" && addr.IsGlobalUnicast() {
-			node.GlobalUnicastIP = addr.String()
-		}
-		if addr.To4() != nil {
-			node.ipv4Support = true
-		} else {
-			node.ipv6Support = true
-		}
+// DiscoverIPMode discovers the IP Versions supported by Proxy based on its IP addresses.
+func (node *Proxy) DiscoverIPMode() {
+	if networkutil.AllIPv4(node.IPAddresses) {
+		node.ipMode = IPv4
+	} else if networkutil.AllIPv6(node.IPAddresses) {
+		node.ipMode = IPv6
+	} else {
+		node.ipMode = Dual
 	}
+	node.GlobalUnicastIP = networkutil.GlobalUnicastIP(node.IPAddresses)
 }
 
 // SupportsIPv4 returns true if proxy supports IPv4 addresses.
 func (node *Proxy) SupportsIPv4() bool {
-	return node.ipv4Support
+	return node.ipMode == IPv4 || node.ipMode == Dual
 }
 
 // SupportsIPv6 returns true if proxy supports IPv6 addresses.
 func (node *Proxy) SupportsIPv6() bool {
-	return node.ipv6Support
+	return node.ipMode == IPv6 || node.ipMode == Dual
```

Comment:
I think `node.ipMode == IPv4 || node.ipMode == Dual` and `node.ipMode == IPv6 || node.ipMode == Dual` should not co-exist at the same time, otherwise, there will be a contradiction, especially when separately using `SupportsIPv4()` or `SupportsIPv6()`

Reply:
SupportsIPv4() just tells whether proxyIP list has atleast one ipv4 address. Same for ipv6